doc/patch/patch-CI-IO.patch

```diff
@@ -0,0 +1,150 @@
+From 23e6b0da8dbdf3f0c6fb5e962d3bc753eb6d078d Mon Sep 17 00:00:00 2001
+From: Tim Griesbach <tim.griesbach@uni-bonn.de>
+Date: Wed, 15 Jan 2025 15:23:29 +0100
+Subject: [PATCH] Implement sleep and retry in case of busy file system
+
+---
+ src/sc_io.c | 92 +++++++++++++++++++++++++++++++++++++++++++++++++++++
+ 1 file changed, 92 insertions(+)
+
+diff --git a/src/sc_io.c b/src/sc_io.c
+index 4cfc7b81..fb3f065c 100644
+--- a/src/sc_io.c
++++ b/src/sc_io.c
+@@ -24,6 +24,13 @@
+ #include <sc_io.h>
+ #include <sc_puff.h>
+ #include <libb64.h>
++#ifdef _MSC_VER
++#include <windows.h>
++#elif _POSIX_C_SOURCE >= 199309L
++#include <time.h>
++#else
++#include <unistd.h>
++#endif
+ #ifdef SC_HAVE_ZLIB
+ #include <zlib.h>
+ #else
+@@ -37,6 +44,60 @@
+ #include <errno.h>
+ #endif
+ 
++/** In case of an unexpected retrieved count 0, this macro sleeps and retries.
++ * In the Github Actions CI, we experienced transient CI failures due to the
++ * file system not performing requested I/O operations.
++ *
++ * The macro can be only callled in the case that expected count is unequal 0.
++ * It checks if count is 0 and if this is true, it sleeps and then retries the
++ * I/O operation once.
++ */
++#define SC_IO_SLEEP_AND_RETRY(func, time) do {\
```

Comment:
I'd be leaning to prefer a function over a macro, quite generally. Would this be practicable here?

Reply:
It is possible but would come with significantly more parameters. Namely, we would need `mpifile`, `offset`, `ptr`, `count`, `t` and `errcode` as additional parameters. What do you think about the trade off between using a macro but reducing the number of parameters from 8 to 2?

Reply:
Agreed, the use of the macro is rather elegant. I'm not sure if we should return out of it, would it make sense to return on the outside, or would that rather complicate things?

Reply:
Yes, I adjusted the macro to return outside of it.